File: src/main/java/de/dc/planetsudo/level/levelobjects/BurningMothershipEffect.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

package de.dc.planetsudo.level.levelobjects;

import java.util.TimerTask;

/**
 *
 * @author divine
 */
public class BurningMothershipEffect extends TimerTask {

	public final Mothership mothership;

	public BurningMothershipEffect(Mothership mothership) {
		this.mothership = mothership;
	}

	@Override
	public void run() {
		mothership.orderFuel(33-mothership.getShieldForce(), null);
	}

}

File: src/main/kotlin/org/openbase/planetsudo/level/levelobjects/BurningMothershipEffect.kt
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package org.openbase.planetsudo.level.levelobjects

import java.util.*

/**
 *
 * @author [Divine Threepwood](mailto:divine@openbase.org)
 */
class BurningMothershipEffect(val mothership: Mothership) : TimerTask() {
    override fun run() {
        mothership.orderFuel(33 - mothership.shieldForce, null)
    }
}
